pkg/providers/tinkerbell/workers.go

```diff
@@ -24,20 +24,12 @@ type (
 // It talks to the cluster with a client to detect changes in immutable objects and generates new
 // names for them.
 func WorkersSpec(ctx context.Context, logger logr.Logger, client kubernetes.Client, spec *cluster.Spec) (*Workers, error) {
-	wgc := spec.Cluster.Spec.WorkerNodeGroupConfigurations
-	workerTemplateNames := make(map[string]string, len(wgc))
-	kubeadmTemplateNames := make(map[string]string, len(wgc))
-
-	for _, wc := range wgc {
-		workerTemplateNames[wc.Name] = clusterapi.WorkerMachineTemplateName(spec, wc)
-		kubeadmTemplateNames[wc.Name] = clusterapi.DefaultKubeadmConfigTemplateName(spec, wc)
-	}
-
 	templateBuilder, err := generateTemplateBuilder(spec)
 	if err != nil {
 		return nil, errors.Wrap(err, "generating tinkerbell template builder")
 	}
 
+	workerTemplateNames, kubeadmTemplateNames := clusterapi.InitialTemplateNamesForWorkers(spec)
```

Comment:
The same logic was duplicated across all our applicable providers thus far for the controller